opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -13,8 +13,158 @@
 # limitations under the License.
 
 
-# TODO: #2247
-# pylint: disable=no-self-use
+from fnmatch import fnmatch
+from logging import getLogger
+from typing import Optional, Set, Type
+
+# FIXME import from typing when support for 3.6 is removed
+from typing_extensions import final
+
+from opentelemetry._metrics.instrument import Instrument
+from opentelemetry.sdk._metrics.aggregation import _AggregationFactory
+
+_logger = getLogger(__name__)
+
+
 class View:
-    def match(self) -> bool:
-        return False
+    def __init__(
+        self,
+        instrument_type: Optional[Type[Instrument]] = None,
+        instrument_name: Optional[str] = None,
+        meter_name: Optional[str] = None,
+        meter_version: Optional[str] = None,
+        meter_schema_url: Optional[str] = None,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        attribute_keys: Set[str] = None,
+        aggregation: Optional[_AggregationFactory] = None,
+    ):
+        """
+        A `View` configuration parameters can be used for the following
+        purposes:
+
+        1. Match instruments: When an instrument matches a view, measurements
+           received by that instrument will be processed.
+        2. Customize metric streams: A metric stream is identified by a match
+           between a view and an instrument and a set of attributes. The metric
+           stream can be customized by certain attributes of the corresponding
+           view.
+
+        The attributes documented next serve one of the previous two purposes.
+
+        Args:
+            instrument_type: This is an instrument matching attribute: the
+                class the instrument must be to match the view.
+
+            instrument_name: This is an instrument matching attribute: the name
+                the instrument must have to match the view. Wild card
+                characters are supported. Wild card characters should not be
+                used with this attribute if the view has also a
+                ``name`` defined.
+
+            meter_name: This is an instrument matching attribute: the name
+                the instrument meter must have to match the view. Wild card
+                characters are supported.
+
+            meter_version : This is an instrument matching attribute: the
+                version the instrument meter must have to match the view.
+                Wild card characters are supported.
```

Comment:
Looking at the spec description for selection criteria I think only instrument name should support the wildcard chars?

Reply:
Right, fixed :v:

Reply:
Fixed